AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/DataStoreModelsVersionTests.swift

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+class DataStoreModelsVersionTests: XCTestCase {
+
+    func testVersionIsEmpty() {
+        let schemaUpdateInvoked = expectation(description: "Version has updated and old database is deleted")
+
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionIsEmpty") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+
+        let newVersion = "newVersion"
+
+        let mockFileManager = MockFileManager()
```

Comment:
you can have an expectation with an inverted fulfill, so it fails if it fulfills
`removeItemInvoked.isInverted = true`

which means you are sure that the file is not deleted

Reply:
Sorry, I don't understand here, one comment below says to remove the wait(for: [schemaUpdateInvoked], timeout: 10), so that expectation is not needed here?

Reply:
hmm this comment is outdated now since code has been changed, let me try to explain again in the new code